### PR TITLE
Add in more NIFM functionality

### PR DESCRIPTION
**Pull request by shibbo**
This pull request adds in a few more functions related to the nifm sysmodule. It allows checking to see if a system is using wireless / ethernet connections, and it can halt / resume all connections.

### CODE REVIEW

File: nx/source/services/nifm.c [hunk 1]
```diff
@@ -2,6 +2,7 @@
  * @file nifm.h
  * @brief Network interface service IPC wrapper.
  * @author shadowninja108
+ * @author shibboleet
```
Comment (yellows8) [inline nx/source/services/nifm.c]: Move to prev line.

File: nx/source/services/nifm.c [hunk 2]
```diff
@@ -81,6 +82,177 @@ Result nifmGetCurrentIpAddress(u32* out) {
...
+            u64 magic;
+            u64 result;
+            u8 out;
+        } *resp;
+
+        serviceIpcParse(&g_nifmIGS, &r, sizeof(*resp));
+        resp = r.Raw;
+
+        rc = resp->result;
+        *out = resp->out;
```
Comment (yellows8) [inline nx/source/services/nifm.c]: newline, and: `if (R_SUCCEEDED(rc) && out) *out = resp->out!=0;` (same below)

File: nx/source/services/nifm.c [hunk 3]
```diff
@@ -21,7 +21,7 @@ Result nifmInitialize(void) {
 
     if (serviceIsActive(&g_nifmSrv))
         return 0;
-		
+
```
Comment (yellows8) [inline nx/source/services/nifm.c]: Replace with newline, likewise elsewhere.

File: nx/include/switch/services/nifm.h [hunk 4]
```diff
@@ -2,6 +2,7 @@
  * @file nifm.h
  * @brief Network interface service IPC wrapper.
  * @author shadowninja108
+ * @author shibboleet
```
Comment (yellows8) [inline nx/include/switch/services/nifm.h]: Merge with prev line.

File: nx/source/services/nifm.c (hunk 3 above)
Comment (yellows8) [reply nx/source/services/nifm.c]: 'Replace with newline, likewise elsewhere.'

Comment (shibbo) [reply nx/source/services/nifm.c]: Sorry, didn't spot that one.

File: nx/source/services/nifm.c [hunk 5]
```diff
@@ -1,7 +1,7 @@
 /**
- * @file nifm.h
+ * @file nifm.c
  * @brief Network interface service IPC wrapper.
- * @author shadowninja108
+ * @author shadowninja10, shibboleet
```
Comment (yellows8) [inline nx/source/services/nifm.c]: Restore the '8' after '10'.

File: nx/source/services/nifm.c [hunk 6]
```diff
@@ -1,7 +1,7 @@
 /**
- * @file nifm.h
+ * @file nifm.c
  * @brief Network interface service IPC wrapper.
- * @author shadowninja108
+ * @author shadowninja108, shibboleet
```
Comment (yellows8) [inline nx/source/services/nifm.c]: Remove whitespace at the end.

File: nx/source/services/nifm.c (hunk 2 above)
Comment (yellows8) [reply nx/source/services/nifm.c]: ping ^

Comment (shibbo) [reply nx/source/services/nifm.c]: Do you want the line on 112 removed?

Comment (yellows8) [reply nx/source/services/nifm.c]: no https://github.com/switchbrew/libnx/pull/236/files#r250879994

**Pull request merged**